internal/apps/config/config.go

```diff
@@ -0,0 +1,116 @@
+package config
+
+import (
+	"strings"
+	"time"
+)
+
+// ConfigSource is a config source.
+type ConfigSource interface {
+	IsSet(key string) bool
+	GetString(key string) string
+	GetBool(key string) bool
+	GetDuration(key string) time.Duration
+}
+
+type mutatingConfigSource struct {
+	cs          ConfigSource
+	mutateKey   func(key string) string
+	mutateIsSet bool // some config sources except IsSet to _not_ include the namespace
```

Comment:
I think this is a bit confusing to have differing access patterns based on this flag; I'd vote we always use the `mutateKey` function if it exists for a mutating source